Ladybird/FontPlugin.cpp

```diff
@@ -50,6 +61,73 @@ Gfx::Font& FontPlugin::default_fixed_width_font()
     return *m_default_fixed_width_font;
 }
 
+#ifdef USE_FONTCONFIG
+static Optional<String> query_fontconfig_for_generic_family(Web::Platform::GenericFont generic_font)
+{
+    char const* pattern_string = nullptr;
+    switch (generic_font) {
+    case Web::Platform::GenericFont::Cursive:
+        pattern_string = "cursive";
+        break;
+    case Web::Platform::GenericFont::Fantasy:
+        pattern_string = "fantasy";
+        break;
+    case Web::Platform::GenericFont::Monospace:
+        pattern_string = "monospace";
+        break;
+    case Web::Platform::GenericFont::SansSerif:
+        pattern_string = "sans-serif";
+        break;
+    case Web::Platform::GenericFont::Serif:
+        pattern_string = "serif";
+        break;
+    case Web::Platform::GenericFont::UiMonospace:
+        pattern_string = "monospace";
+        break;
+    case Web::Platform::GenericFont::UiRounded:
+        pattern_string = "sans-serif";
+        break;
+    case Web::Platform::GenericFont::UiSansSerif:
+        pattern_string = "sans-serif";
+        break;
+    case Web::Platform::GenericFont::UiSerif:
+        pattern_string = "serif";
+        break;
+    default:
+        VERIFY_NOT_REACHED();
+    }
+
+    auto* config = FcConfigGetCurrent();
+
+    FcPattern* pattern = FcNameParse(reinterpret_cast<FcChar8 const*>(pattern_string));
+    FcConfigSubstitute(config, pattern, FcMatchPattern);
+    FcDefaultSubstitute(pattern);
+
+    // Never select bitmap fonts.
+    FcPatternAddBool(pattern, FC_SCALABLE, FcTrue);
+
+    // FIXME: Enable this once we can handle OpenType variable fonts.
+    FcPatternAddBool(pattern, FC_VARIABLE, FcFalse);
+
+    FcResult result;
```

Comment:
surely we want to initialize this and `family` below with either `= 0`, `= nullptr` or `= {}` ?